app/grandchallenge/algorithms/views.py

```diff
@@ -753,3 +759,50 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context.update({"algorithm": self.base_object})
         return context
+
+
+class AlgorithmAddRepo(
+    LoginRequiredMixin, ObjectPermissionRequiredMixin, UpdateView,
+):
+    model = Algorithm
+    form_class = AlgorithmRepoForm
+    template_name = "algorithms/algorithm_add_repo.html"
+    permission_required = (
+        f"{Algorithm._meta.app_label}.change_{Algorithm._meta.model_name}"
+    )
+    raise_exception = True
+
+    def get_form_kwargs(self):
+        """Return the keyword arguments for instantiating the form."""
+        kwargs = super().get_form_kwargs()
+
+        code = self.request.GET.get("code")
+        repos = []
+        if code is not None:
+            headers = {"Accept": "application/vnd.github.v3+json"}
+
+            resp = requests.post(
+                "https://github.com/login/oauth/access_token",
+                data={
+                    "code": code,
```

Comment:
Do we need to clean `code` at all?

Reply:
I think this should be ok as it is a query param and should be url encoded? Either way, it only gets posted to github, so worst case scenario for us is that we don't get the proper response I think.